### ISSUE TITLE
test(anomaly): use a real anomaly detection dataset for unit tests.

### ISSUE THREAD

**Issue by tachyonicClock** [adaptive-machine-learning/CapyMOA#414]
Currently we re-purpose `ElectricityTiny` as an anomaly detection dataset. This does not work well because it has a much higher concentration of "anomalous" exemplars, making them not true anomalies. This means that the detector misses what is labelled as an anomaly because it's not a true anomaly.

We should switch to using a true anomaly detection dataset. Ideally this would be a lightweight dataset similar to `ElectricityTiny`.

Identified in #413 thanks @mateenali66

**Comment by hmgomes** [comment]
We can just use a simple dataset based on `make_blobs` like in the [anomaly detection tutorial](https://capymoa.org/notebooks/anomaly_detection.html), see below a slightly modified version from the tutorial

```
import numpy as np
from sklearn.datasets import make_blobs
import matplotlib.pyplot as plt
from capymoa.stream import NumpyStream
from capymoa.anomaly import HalfSpaceTrees
from capymoa.evaluation import AnomalyDetectionEvaluator

# generate normal data points
n_samples = 10000
n_features = 2
n_clusters = 3
X, y = make_blobs(
    n_samples=n_samples, n_features=n_features, centers=n_clusters, random_state=42
)

# generate anomalous data points
n_anomalies = 50  # the anomaly rate is 0.5%
anomalies = np.random.uniform(low=-10, high=10, size=(n_anomalies, n_features))

# combine the normal data points with anomalies
X = np.vstack([X, anomalies])
y = np.hstack([y, [1] * n_anomalies])  # Label anomalies with 1
y[:n_samples] = 0  # Label normal points with 0

# shuffle the data
idx = np.random.permutation(n_samples + n_anomalies)
X = X[idx]
y = y[idx]

plt.scatter(X[:, 0], X[:, 1], c=y, cmap="viridis")
plt.show()

stream_ad = NumpyStream(
    X,
    y,
    dataset_name="AnomalyDetectionDataset",
    feature_names=[f"feature_{i}" for i in range(n_features)],
    target_name="class",
    target_type="categorical",
)
learner = HalfSpaceTrees(stream_ad.get_schema())
evaluator = AnomalyDetectionEvaluator(stream_ad.get_schema())
while stream_ad.has_more_instances():
    instance = stream_ad.next_instance()
    score = learner.score_instance(instance)
    evaluator.update(instance.y_index, score)
    learner.train(instance)

auc = evaluator.auc()
print(f"AUC: {auc:.2f}")
```

**Comment by mateenali66** [comment]
Taking this one.

Ran your snippet and sized it against the current fixture. Three things worth settling before I open a PR.

**Size.** 10,000 normal plus 50 anomalies costs 69s for the nine non-torch detectors, against 12.6s for `ElectricityTiny` today. At 2,000 normal it is 15s, which matches the lightweight fixture the issue asks for. Proposing 2,000.

**Anomaly count.** 50 anomalies at 0.5 percent leaves too few positives to pin an AUC. Spread over five data seeds at 2,000 normal:

| anomalies | detectors with seed spread under 0.06 |
|---|---|
| 10 | 0 of 9 |
| 100 | 5 of 9 |

At 100, `StreamingIsolationForest`, `RobustRandomCutForest`, `AdaptiveIsolationForest`, `IForestASD` and `RSHash` all sit within 0.05. Proposing 100, which is 4.8 percent rather than a realistic rate, on the grounds that these are code-path tests rather than benchmarks.

**`OnlineIsolationForest`.** @tachyonicClock was right on #413 and the dataset was hiding it. The test's `window_size=100` goes below chance on 2 of 5 data seeds. The library defaults do not.

| config | five data seeds | min | time |
|---|---|---|---|
| test, ws=100 | 0.811 0.665 0.478 0.607 0.360 | 0.360 | 3.6s |
| library defaults | 0.869 0.856 0.752 0.820 0.651 | 0.651 | 6.4s |

On `ElectricityTiny` no configuration reached 0.5, because the label is price direction rather than an anomaly. On real anomalies the defaults never drop below 0.65. So the fixture swap needs that config change with it, otherwise the `> 0.5` floor from #415 fails.

Happy to send it as proposed, or to stay closer to the tutorial on size and rate if you would rather.

**Comment by hmgomes** [comment]
Thanks @mateenali66 for your comments and for bringing this up. #418 closes this issue
